main_test.go

```diff
@@ -87,6 +89,82 @@ func TestValidate(t *testing.T) {
 	}
 }
 
+func TestDefault(t *testing.T) {
+	flag.Parse()
+	want := `{
+  "xds_servers": [
+    {
+      "server_uri": "trafficdirector.googleapis.com:443",
+      "channel_creds": [
+        {
+          "type": "google_default"
+        }
+      ],
+      "server_features": [
+        "xds_v3"
+      ]
+    }
+  ],
+  "authorities": {
+    "": {}
+  },
+  "node": {
+    "id": "projects/0/networks/default/nodes/31313131-3131-4131-b131-313131313131",
+    "cluster": "cluster",
+    "metadata": {
+      "INSTANCE_IP": "127.0.0.1",
+      "TRAFFICDIRECTOR_GCP_PROJECT_NUMBER": "0",
+      "TRAFFICDIRECTOR_NETWORK_NAME": "default"
+    },
+    "locality": {
+      "zone": "us-central-1"
+    }
+  },
+  "certificate_providers": {
+    "google_cloud_private_spiffe": {
+      "plugin_name": "file_watcher",
+      "config": {
+        "certificate_file": "/var/run/secrets/workload-spiffe-credentials/certificates.pem",
+        "private_key_file": "/var/run/secrets/workload-spiffe-credentials/private_key.pem",
+        "ca_certificate_file": "/var/run/secrets/workload-spiffe-credentials/ca_certificates.pem",
+        "refresh_interval": "600s"
+      }
+    }
+  },
+  "server_listener_resource_name_template": "grpc/server?xds.resource.listening_address=%s"
+}`
+	input := configInput{
+		xdsServerUri:               *xdsServerUri,
```

Comment:
I was exploring ways to test what the default behavior of the bootstrap generator is. Particularly what happens when the default values of the flags are applied to generate config. 

I agree that this method is not trustworthy at the moment, and would require manual verification for every PR that adds/updates a flag by a maintainer. But having a test like this would allow us to assert things like Federation is enabled by default or like in a future change -- meshID is *not* auto generated by default. 

When i look at ways to test main() it wasnt obvious unless we do some refactoring of the code + mock a few APIs. 

If you strongly feel that this method is not sustainable, I can drop this PR and later look at a refactor which would allow us to test the default behavior more easier. Are you not convinced that this is the default behavior. Let me know. @ejona86